### PR TITLE
Add config options for max file upload sizes

### LINKED ISSUE #2198: Allow configuring the max upload size for files
### Discussed in https://github.com/pretix/pretix/discussions/2188

<div type='discussions-op-text'>

<sup>Originally posted by **neumantm** September  1, 2021</sup>
I would  like to set it from `pretix.cfg`.
The current hard coded value could be used as the default.</div>

<div type='discussions-op-text'>

<sup>Originally posted by **rphaelm** September  2, 2021</sup>
The 10 MB limit for file upload is in different places throughout the codebase, see e.g. this search:

https://github.com/pretix/pretix/search?q=max_size

### PR DESCRIPTION
**Pull request by neumantm**
Closes #2198

### CODE REVIEW

File: doc/admin/config.rst [hunk 1]
```diff
@@ -434,3 +434,19 @@ pretix can make use of some external tools if they are installed. Currently, the
...
+Maximum upload file sizes
+-------------------------
+
+You can configure the maximum file size for uploading various files::
+
+    [pretix_file_upload]
+    ; Max upload size for images in MiB, defaults to 10 MiB
+    max_size_image = 12
+    ; Max upload size for favicons in MiB, defaults to 1 MiB
+    max_size_image = 2
```
Comment (wiffbi) [inline doc/admin/config.rst:447]:
```suggestion
    max_size_favicon = 2
```

File: doc/admin/config.rst [hunk 2]
```diff
@@ -434,3 +434,19 @@ pretix can make use of some external tools if they are installed. Currently, the
...
+    ; Max upload size for email attachments in MiB, defaults to 10 MiB
+    max_size_image = 15
```
Comment (wiffbi) [inline doc/admin/config.rst:449]:
```suggestion
    max_size_email_attachment = 15
```

File: doc/admin/config.rst [hunk 3]
```diff
@@ -434,3 +434,19 @@ pretix can make use of some external tools if they are installed. Currently, the
...
+    ; Max upload size for other files in MiB, defaults to 10 MiB
+    ; This includes all file upload type order questions
+    max_size_image = 100
```
Comment (wiffbi) [inline doc/admin/config.rst:452]:
```suggestion
    max_size_other = 100
```

**Review by wiffbi — CHANGES_REQUESTED** [review]
I think this is a very useful addition to pretix! Thank you!

File: src/pretix/settings.py [hunk 4]
```diff
@@ -845,4 +845,10 @@ def traces_sampler(sampling_context):
 DATA_UPLOAD_MAX_NUMBER_FIELDS = 25000
 DATA_UPLOAD_MAX_MEMORY_SIZE = 10 * 1024 * 1024  # 10 MB
 
+# File sizes are in MB
```
Comment (raphaelm) [inline src/pretix/settings.py:848]: Thecomment says MB, the new docs say MiB. I always get confused with the two, which is correct here and should we use it consistently? ;)

Comment (wiffbi) [reply src/pretix/settings.py:848]: I guess it is MiB due to the 1024*1024 instead of 1000*1000

File: doc/admin/config.rst (hunk 3 above)
Comment (neumantm) [reply doc/admin/config.rst:452]: oops

File: src/pretix/settings.py (hunk 4 above)
Comment (neumantm) [reply src/pretix/settings.py:848]:
```suggestion
# File sizes are in MiB
```

Other occurrences of MiB in this file are also wrong.
But I think as MB and MiB are not that far apart, it is not such a big problem.

**Comment by raphaelm** [comment]
Thank you! :sparkles:

**Pull request merged**